microcosm_resourcesync/endpoints/directory_endpoint.py

```diff
@@ -44,7 +44,9 @@ def read(self, schema_cls, **kwargs):
                 _, ext = splitext(filename)
                 formatter = Formatters.for_extension(ext).value
                 with open(join(dirpath, filename), "r") as file_:
-                    yield schema_cls(formatter.load(file_.read()))
+                    data = file_.read()
```

Comment:
I split this into steps because I was evaluating the performance of each step. The `formatter.load(data)` call was the big one.